controllers/secretagentconfiguration_controller.go

```diff
@@ -208,21 +209,35 @@ func labelsForSecretAgent(name string) map[string]string {
 	return map[string]string{"managed-by-secret-agent": "true", "secret-agent-configuration-name": name}
 }
 
+// Update the SecretAgentConfiguration status with the object names
 func (reconciler *SecretAgentConfigurationReconciler) updateStatus(ctx context.Context, instance *v1alpha1.SecretAgentConfiguration, inProgress, errorFound bool) error {
-	// Update the SecretAgentConfiguration status with the object names
-	secretList := &corev1.SecretList{}
+	var totalManagedObjects int
+	var secretNames []string
+	mismatchedNumSecrets := false
+	maxRetries := 5 //wait maximum of 1sec for the API to update
 	listOpts := []client.ListOption{
 		client.InNamespace(instance.Namespace),
 		client.MatchingLabels(labelsForSecretAgent(instance.Name)),
 	}
-	if err := reconciler.List(ctx, secretList, listOpts...); err != nil {
-		return err
-	}
-	var secretNames []string
-	for _, secret := range secretList.Items {
-		secretNames = append(secretNames, secret.Name)
+	// retry up to maxRetries. Wait 200ms after each iteration
+	for it := 0; it < maxRetries; it++ {
+		secretNames = nil
+		secretList := &corev1.SecretList{}
+		if err := reconciler.List(ctx, secretList, listOpts...); err != nil {
+			return err
+		}
+		for _, secret := range secretList.Items {
+			secretNames = append(secretNames, secret.Name)
+		}
+		totalManagedObjects = len(secretNames) // TODO Need to include cloud stored secrets in SAC status
+		if len(instance.Spec.Secrets) == len(secretList.Items) {
+			mismatchedNumSecrets = false
+			break
+		}
+		mismatchedNumSecrets = true
+		time.Sleep(200 * time.Millisecond)
```

Comment:
Hmm - any way we can avoid this?

Is the objective to just update the status?  Maybe we can walk through it - but can we just ask to be reconciled again?